contracts/models/contract_enterprise.py

```diff
@@ -0,0 +1,50 @@
+from odoo import api, fields, models, tools, _
+from odoo.exceptions import UserError, ValidationError
+from datetime import datetime, timedelta
+
+class ContractEnterprise(models.Model):
+    _name = "contract.enterprise"
+    _description = "contract enterprise model"
+
+    def _default_sign_date(self):
+        next_week = fields.Datetime.now() + timedelta(weeks=1)
+        if next_week.weekday() >= 5:
+            next_week += timedelta(days=7 - next_week.weekday())
+        return next_week
+
+    id = fields.Integer('id', required=True)
+    name = fields.Char("contract name", required=True)
+    start_date = fields.Datetime('start date', required=True, select=True, default=datetime.now())
+    end_date = fields.Datetime('end date', required=True, select=True)
+    type_contract = fields.Integer('contract type', required=True)
+    sign_date = fields.Datetime('sign date', required=True, select=True, default=_default_sign_date)
+    salary_level = fields.Float('salary level', required=False, default=0)
+    effective_salary = fields.Float('effective salary', required=False, default=0)
+    employee_id = fields.Many2one('employee.directory', string="Employee", required=True, ondelete='cascade')
+    '''
+        ondelete : có 4 type
+            + cascade : khi một bản ghi được xóa, tất cả các bản ghi liên quan trong table con sẽ cũng được xóa.
+            + set null :  khi một bản ghi được xóa, trường Many2one sẽ được đặt thành giá trị NULL.
+            + restrict : Odoo sẽ không cho phép xóa bản ghi nếu có bản ghi liên quan trong bảng con.
+            + no action : Odoo sẽ không thực hiện bất kỳ hành động nào khi một bản ghi được xóa. Bạn phải quản lý các hành động liên quan bằng cách thủ công
+    '''
+    sign_day = fields.Date(string="Sign Date")
+    status = fields.Selection([
+                ('new', 'New'),
+                ('running', 'Running'),
+                ('expired', 'Expired'),
+                ('pause', 'Pause')
+            ], string='Status', default='new')
+
+    total_salary = fields.Float(string='Total Salary', compute='_compute_total_salary', store=True)
+
+    @api.depends('salary_level', 'effective_salary')
```

Comment:
@Tampham2-ptt depends ở đây có ý nghĩa là gì?

Reply:
+ depends và onchange có mục đích sử dụng giống nhau là được sử dụng để : khi các giá trị được mô tả('salary_level', 'effective_salary') có sự thay đổi về giá trị thì function sẽ được kích hoạt thực hiển để tính toán lại giá trị .
+ tuy nhiên depend và onchange có sự khác nhau là:
	- api.onchange : để tạo ra 1 giá trị ảo mà không cập nhật vào DB, thực hiện thay đổi ngay lập tức khi có sự thay đổi từ trường chỉ định
	- api.depend : để tạo ra 1 field là sự tính toán từ các field khác và cập nhật vào DB, sẽ thay đổi dữ liệu khi submit, không thực hiện thay đổi ngay

Reply:
sự khác nhau chưa đúng lắm đọc lại tài liệu của Odoo em nhé

Reply:
sự khác nhau giữa depends và onchange
+ onchange: 
   - chỉ hỗ trợ các trường đơn giản, không hỗ trợ trường phụ thuộc (ví dụ test.name)
   - chỉ bắt event trên view (ví dụ giao diện form)
   - Trường onchange có mặc định store = True
   - Không cần khai báo trên field model
 + compute:
   - hỗ trợ trường phụ thuộc (có thể nằm ở model khác)
   - Trường compute có mặc định store = False
   - Cần khai báo hàm compute trên field model qua thuộc tính compute